### PR TITLE
Feature/optimize log softmax operator

### PR DESCRIPTION
**Pull request by liuzm6217-jianan**

### CODE REVIEW

File: src/Native/src/kernels/stackvm/optimized/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ set(SRCS concat.cpp
          gather_nd.cpp
          quantize.cpp
          onehot.cpp
+		 # log_softmax.cpp
```
Comment (sunnycase) [inline src/Native/src/kernels/stackvm/optimized/CMakeLists.txt:24]: Fix ident.

File: src/Native/src/kernels/stackvm/optimized/CMakeLists.txt [hunk 2]
```diff
@@ -49,6 +50,7 @@ _TARGET_ARCH_FILES(TARGET kernels
                    softmax.cpp
                    unary.cpp
                    where.cpp
+				   log_softmax.cpp
```
Comment (sunnycase) [inline src/Native/src/kernels/stackvm/optimized/CMakeLists.txt:53]: Fix ident.

Comment (sunnycase) [inline src/Native/src/kernels/stackvm/optimized/CMakeLists.txt:53]: Fix ident.

File: src/Native/src/kernels/stackvm/tensor_ops.cpp [hunk 3]
```diff
@@ -385,9 +385,9 @@ result<value_t> nncase::kernels::stackvm::log_softmax(
     try_f32_input(in_mem, input);
     try_f32_output(out_mem, output, input_tensor->shape());
     try_positive_axis(axis_value, axis, input_tensor);
-    try_(reference::softmax(in_mem, out_mem, input_tensor->shape(),
-                            input_tensor->strides(), output_tensor->strides(),
-                            axis_value, 1.f, true));
+    try_(optimized::log_softmax(in_mem, out_mem, input_tensor->shape(),
```
Comment (sunnycase) [inline src/Native/src/kernels/stackvm/tensor_ops.cpp:388]: Should use contiguous kernel helper as the optimized kernel doesn't handle incontiguous inputs.

File: src/Native/src/kernels/stackvm/optimized/CMakeLists.txt (hunk 1 above)
Comment (liuzm6217-jianan) [reply src/Native/src/kernels/stackvm/optimized/CMakeLists.txt]: Done

File: src/Native/src/kernels/stackvm/optimized/CMakeLists.txt (hunk 2 above)
Comment (liuzm6217-jianan) [reply src/Native/src/kernels/stackvm/optimized/CMakeLists.txt]: Done

Comment (liuzm6217-jianan) [reply src/Native/src/kernels/stackvm/optimized/CMakeLists.txt]: Done!

File: src/Native/src/kernels/stackvm/tensor_ops.cpp (hunk 3 above)
Comment (liuzm6217-jianan) [reply src/Native/src/kernels/stackvm/tensor_ops.cpp]: Done1

File: src/Native/src/kernels/stackvm/tensor_ops.cpp [hunk 4]
```diff
@@ -385,9 +385,17 @@ result<value_t> nncase::kernels::stackvm::log_softmax(
...
-                            input_tensor->strides(), output_tensor->strides(),
-                            axis_value, 1.f, true));
+    // reference log_softmax have no define, so now cannot use
+    // CONTIGUOUS_KERNEL macro
+    if (is_contiguous(input_tensor)) {
+        try_(optimized::log_softmax(in_mem, out_mem, input_tensor->shape(),
+                                    input_tensor->strides(),
+                                    output_tensor->strides(), axis_value));
+    } else {
+        try_(reference::softmax(
```
Comment (sunnycase) [inline src/Native/src/kernels/stackvm/tensor_ops.cpp:395]: Should add reference log_softmax implementation.

Comment (liuzm6217-jianan) [reply src/Native/src/kernels/stackvm/tensor_ops.cpp]: Done!

**Review by FusionBolt — APPROVED** [review]

**Pull request merged**